astroquery/alma/core.py

```diff
@@ -79,140 +255,167 @@ def query_region_async(self, coordinate, radius, cache=True, public=True,
-                    log.info("Failed query.  Retrying up to {0} more times"
-                             .format(max_retries))
-                    return self.query_async(payload=payload, cache=False,
-                                            public=public, science=science,
-                                            max_retries=max_retries-1,
-                                            get_html_version=get_html_version,
-                                            get_query_payload=get_query_payload,
-                                            **kwargs)
-                raise RemoteServiceError("Incorrect return from HTML table query.")
-            response2 = self._request('GET',
-                                      "{0}/{1}/{2}".format(
-                                          self._get_dataarchive_url(), 'aq',
-                                          response.text),
-                                      params={'query_url':
-                                              response.url.split("?")[-1]},
-                                      timeout=self.TIMEOUT,
-                                      cache=False,
-                                      )
-            self._last_response = response2
-            response2.raise_for_status()
-            if len(response2.text) == 0:
-                if max_retries > 0:
-                    log.info("Failed (empty) query.  Retrying up to {0} more times"
-                             .format(max_retries))
-                    return self.query_async(payload=payload, cache=cache,
-                                            public=public, science=science,
-                                            max_retries=max_retries-1,
-                                            get_html_version=get_html_version,
-                                            get_query_payload=get_query_payload,
-                                            **kwargs)
-                raise RemoteServiceError("Empty return.")
-            return response2
+            payload['science_observations'] = True
 
+        if public is not None:
+            if public:
+                payload['public_data'] = True
+            else:
+                payload['public_data'] = False
+        query = _gen_sql(payload)
+        result = self.query_tap(query, **kwargs)
+        if result:
+            result = result.to_table()
         else:
-            return response
-
-    def validate_query(self, payload, cache=True):
-        """
-        Use the ALMA query validator service to check whether the keywords are
-        valid
+            return result
+        # add 'Observation date' column
+        result['Observation date'] = \
+            [Time(_['t_min'], format='mjd').strftime(ALMA_DATE_FORMAT)
+                for _ in result]
+        for ii in _OBSCORE_TO_ALMARESULT:
+            if _OBSCORE_TO_ALMARESULT[ii] not in result.columns:
+                # duplicate equivalent columns
```

Comment:
why are we doing this?   For backward compatibility?  I don't like it, I'd rather toss one or the other.  I guess we can toss the old, stick with the new.  We should add, though, a renaming utility using the `_OBSCORE_TO_ALMARESULT` mapping to allow users to easily recover the old naming scheme.